config/deb.am

```diff
@@ -38,9 +43,14 @@ if CONFIG_USER
 	pkg7=$${name}-test-$${version}.$${arch}.rpm; \
 	pkg8=$${name}-dracut-$${version}.$${arch}.rpm; \
 	pkg9=$${name}-initramfs-$${version}.$${arch}.rpm; \
+	path_prepend=`mktemp -d /tmp/intercept.XXX`; \
+	ln -s /bin/true $${path_prepend}/dh_shlibdeps; \
```

Comment:
@loli10K The current method for determining package dependencies relies on alien running `dh_shlibdeps`. This method is fundamentally broken: on systems that do not already have these *particular* packages *already* installed, the Debian function will not generate the correct package dependencies. You can see this by reproducing my bugreport zfsonlinux/zfs#6106 by

1. Uninstalling all zfsonlinux packages
2. Installing the Debian ZFS packages
3. Building the zfsonlinux packages.

Then, the built packages will continue to depend on the Debian packages (!!!)

This is not an unreasonable thing to want to do (say, to migrate from a machine running the Debian version, to a system running upstream version). **Moreover**: if you simply uninstall all zfsonlinux packages, `dh_shlibdeps` will not recognize the libraries (since they are part of no installed package), and will, in fact, generate no dependencies.

I most definitely agree this approach is far from perfect---ideally, a a `debian` directory would be included, and a proper Debian build structure used, with proper dependency handling. However, as it stands, the use of `alien` produces unreliable dependencies---sometimes none, sometimes invalid ones, and sometimes (some of) the correct ones. This results in uninstallable packages.